docs/Technical-Documentation/clean-and-reparse.md

```diff
@@ -0,0 +1,129 @@
+# Clean and Re-parse DataFiles
+
+## Background
+As TDP has evolved so has it's validation mechanisms, messages, and expansiveness. As such, many of the datafiles locked in the database and S3
+have not undergone TDP's latest and most stringent validation processes. Because data quality is so important to all TDP stakeholders
+we wanted to introduce a way to re-parse and subsequently re-validate datafiles that have already been submitted to TDP to enhance the integrity
+and the quality of the submissions. The following lays out the process TDP takes to automate and execute this process, and how this process can
+be tested locally and in our deployed environments.
+
+## Clean and Re-parse Flow
+As a safety measure, this process must ALWAYS be executed manually by a system administrator. Once executed, all processes thereafter are completely
+automated. The steps below outline how this process executes.
+
+1. System admin logs in to the appropriate backend application. E.g. `tdp-backend-raft`.
```

Comment:
because this feature currently requires admins to work from CLI, we'll need more documentation on the commands to run. below are the commands I had to run for step 1. 

```shell=

# 0. disconnect from VPN. 

# 1. authenticate into cloud.gov 
apennington@HHSLBDSWL73 MINGW64 /
$ cf login -a api.fr.cloud.gov  --sso
API endpoint: api.fr.cloud.gov

Temporary Authentication Code ( Get one at https://login.fr.cloud.gov/passcode ): <one-time passcode redacted>

Authenticating...
OK


Select an org:
1. hhs-acf-ofa
2. sandbox-hhs

Org (enter to skip): 1
1
Targeted org hhs-acf-ofa.

Select a space:
1. tanf-dev
2. tanf-prod
3. tanf-staging

Space (enter to skip): 1
1
Targeted space tanf-dev.

API endpoint:   https://api.fr.cloud.gov
API version:    3.170.0
user:           alexandra.pennington@acf.hhs.gov
org:            hhs-acf-ofa
space:          tanf-dev

# 2. ssh into backend app
apennington@HHSLBDSWL73 MINGW64 /
$ cf curl v3/apps/$(cf app tdp-backend-qasp --guid)/processes | jq --raw-output '.resources | .[]? | select(.type == "web").guid'
<PROCESS_GUID redacted>

apennington@HHSLBDSWL73 MINGW64 /
$ cf ssh-code
<ssh code redacted>

apennington@HHSLBDSWL73 MINGW64 /
$ ssh -p 2222 cf:<PROCESS_GUID redacted>/0@ssh.fr.cloud.gov
The authenticity of host '[ssh.fr.cloud.gov]:2222 ([2620:108:d00f::fcd:e8d8]:2222)' can't be established.
RSA key fingerprint is <key redacted>.
This key is not known by any other names
Please type 'yes', 'no' or the fingerprint: yes
Could not create directory '/u/.ssh' (No such file or directory).
Failed to add the host to the list of known hosts (/u/.ssh/known_hosts).
cf:<PROCESS_GUID redacted>/0@ssh.fr.cloud.gov's password:<ssh code will be invisible>
vcap@f8eeb863-ee2b-4b0d-7e7c-bc63:~$

# 3. change to directory where python commands can be run
/tmp/lifecycle/shell

# 4. check re-parse command options
python manage.py clean_and_reparse -h
```